src/page-multi-facility/SyncReferenceFacilityModal.tsx

```diff
@@ -0,0 +1,161 @@
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-between;
+  margin-top: 20px;
+`;
+
+const SyncDataTitleContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const TitleText = styled.h1`
+  font-weight: 500;
+  font-size: 15px;
+  line-height: 22px;
+  color: ${Colors.forest};
+  font-family: "Libre Franklin";
+`;
+
+const Row = styled.div<{ selected: boolean }>`
+  border-bottom: 1px solid ${Colors.paleJade20};
+  color: ${(props) => (props.selected ? Colors.forest : Colors.paleJade)};
+  cursor: pointer;
+  display: flex;
+  flex-flow: row nowrap;
+  font-family: "Libre Franklin", sans serif;
+  font-size: 15px;
+  line-height: 16px;
+  letter-spacing: -0.01em;
+  padding: 24px 0;
+`;
+
+const RowIcon = styled.img`
+  margin-right: 12px;
+`;
+
+const SyncDataTitle = (
+  <SyncDataTitleContainer>
+    <TitleText>Sync Data</TitleText>
+    <TitleText>
+      To sync with daily public facility data, please select the corresponding
+      facility.
+    </TitleText>
+  </SyncDataTitleContainer>
+);
+
+interface Props {
+  facilityId: string | null;
+}
+
+interface ReferenceFacilityProps {
+  referenceFacility: ReferenceFacility;
+  onClick: () => void;
+  selected: boolean;
+}
+
+const ReferenceFacilityRow: React.FC<ReferenceFacilityProps> = ({
+  referenceFacility,
+  onClick,
+  selected,
+}) => {
+  return (
+    <Row selected={selected} onClick={onClick}>
+      <RowIcon
+        src={selected ? dataSyncSelectedIcon : dataSyncIcon}
+        alt="data sync icon"
+      />
+      {referenceFacility.canonicalName}
+    </Row>
+  );
+};
+
+const SyncReferenceFacilityModal: React.FC<Props> = ({ facilityId }) => {
+  const {
+    state: { facilities, referenceFacilities },
+    actions: { deselectFacility },
+  } = useFacilities();
+  const [selectedRefFacility, setSelectedRefFacility] = useState<string | null>(
+    null,
+  );
+  const facility = getFacilityById(facilities, facilityId);
+
+  function handleClick(refFacility: ReferenceFacility) {
+    if (selectedRefFacility !== refFacility.id) {
+      setSelectedRefFacility(refFacility.id);
+    } else {
+      setSelectedRefFacility(null);
+    }
+  }
+
+  function handleSave() {
+    if (selectedRefFacility) {
+      deselectFacility();
+      // save a synced composite facility?
+      console.log({ facility, selectedRefFacility });
+      navigate("/");
+    }
+  }
+
+  return (
+    <ModalDialog open={!!facilityId} title={SyncDataTitle}>
+      <ModalContent>
+        {Object.values(referenceFacilities).map((refFacility) => (
+          <ReferenceFacilityRow
+            key={refFacility.id}
+            selected={selectedRefFacility === refFacility.id}
+            referenceFacility={refFacility}
+            onClick={() => handleClick(refFacility)}
+          />
+        ))}
+      </ModalContent>
+      <ModalFooter>
+        <CancelButton
+          onClick={() => {
+            deselectFacility();
+            navigate("/");
```

Comment:
these two lines are repeated several times across files, they should probably be in some kind of "exit this page" function that can be used by the parent component and passed in here as an "onClose" prop or some such